dev/PowerNotifications/PowerNotifications.cpp

```diff
@@ -0,0 +1,316 @@
+    {
+        BatteryStatus_Unregister();
+    }
+
+    void RemainingChargePercent_Update()
+    {
+        BatteryStatus_Update();
+    }
+
+    // RemainingDischargeTime Functions
+    EventType& RemainingDischargeTime_Event()
+    {
+        return make_self<factory_implementation::PowerManager>()->m_remainingDischargeTimeChangedEvent;
+    }
+
+    void RemainingDischargeTime_Register()
+    {
+        check_hresult(PowerNotifications_RegisterDischargeTimeChangedListener(
+            &PowerManager::RemainingDischargeTimeChanged_Callback,
+            &make_self<factory_implementation::PowerManager>()->m_dischargeTimeHandle));
+    }
+
+    void RemainingDischargeTime_Unregister()
+    {
+        check_hresult(PowerNotifications_UnregisterDischargeTimeChangedListener(
+            make_self<factory_implementation::PowerManager>()->m_dischargeTimeHandle));
+    }
+
+    void RemainingDischargeTime_Update()
+    {
+        check_hresult(PowerNotifications_GetDischargeTime(
+            &make_self<factory_implementation::PowerManager>()->m_cachedDischargeTime));
+    }
+
+    // PowerSourceKind Functions
+    EventType& PowerSourceKind_Event()
+    {
+        return make_self<factory_implementation::PowerManager>()->m_powerSourceKindChangedEvent;
+    }
+
+    void PowerSourceKind_Register()
+    {
+        check_hresult(PowerNotifications_RegisterPowerConditionChangedListener(
+            &PowerManager::PowerSourceKindChanged_Callback,
+            &make_self<factory_implementation::PowerManager>()->m_powerSourceKindHandle));
+    }
+
+    void PowerSourceKind_Unregister()
+    {
+        check_hresult(PowerNotifications_UnregisterPowerConditionChangedListener(
+            make_self<factory_implementation::PowerManager>()->m_powerSourceKindHandle));
+    }
+
+    void PowerSourceKind_Update()
+    {
+        check_hresult(PowerNotifications_GetPowerCondition(
+            &make_self<factory_implementation::PowerManager>()->m_cachedPowerSourceKind));
+    }
+
+    // DisplayStatus Functions
+    EventType& DisplayStatus_Event()
+    {
+        return make_self<factory_implementation::PowerManager>()->m_displayStatusChangedEvent;
+    }
+
+    void DisplayStatus_Register()
+    {
+        check_hresult(PowerNotifications_RegisterDisplayStatusChangedListener(
+            &PowerManager::DisplayStatusChanged_Callback,
+            &make_self<factory_implementation::PowerManager>()->m_displayStatusHandle));
+    }
+
+    void DisplayStatus_Unregister()
+    {
+        check_hresult(PowerNotifications_UnregisterDisplayStatusChangedListener
+        (make_self<factory_implementation::PowerManager>()->m_displayStatusHandle));
+    }
+
+    void DisplayStatus_Update()
+    {
+        check_hresult(PowerNotifications_GetDisplayStatus(
+            &make_self<factory_implementation::PowerManager>()->m_cachedDisplayStatus));
+    }
+
+    // SystemIdleStatus Functions
+    EventType& SystemIdleStatus_Event()
```

Comment:
I see many comments that say what the family of functions do.  Instead of using comments, put the implementations into different files.

Reply:
Putting in different files would break the header which is actually a problem when you have a factory_implementation. It results in more issues than is worthwhile to fix; have tried it. :(